torch/csrc/generic/Tensor.cpp

```diff
@@ -1185,17 +1185,7 @@ static bool THPTensor_(_advancedIndexSelect)(PyObject *index, THTensorPtr &dest,
       index, src, broadcasted, &linearIndices, &flattened);
 
   if (success) {
-    // Verify dest tensor is contiguous before flattening
-    THTensor *contiguous = THTensor_(newContiguous)(LIBRARY_STATE dest);
-    THTensor *cviewed = THTensor_(newWithStorage1d)(LIBRARY_STATE
-                                                    THTensor_(storage)(LIBRARY_STATE contiguous),
-                                                    THTensor_(storageOffset)(LIBRARY_STATE contiguous),
-                                                    THTensor_(nElement)(LIBRARY_STATE contiguous),
-                                                    1);
-
-    THTensor_(indexSelect)(LIBRARY_STATE cviewed, flattened, 0, linearIndices);
-    THTensor_(free)(LIBRARY_STATE contiguous);
-    THTensor_(free)(LIBRARY_STATE cviewed);
+    THTensor_(indexSelect)(LIBRARY_STATE dest, flattened, 0, linearIndices);
```

Comment:
It looks like it can only be triggered by calling _advanced_index_select on a tensor or variable (the double backwards case is the only time that function is called in the library).  Given that I don't think we consider that function part of the public interface, do you still want a test?